homeassistant/components/sensor/solaredge.py

```diff
@@ -0,0 +1,176 @@
+"""
+Support for SolarEdge Monitoring API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.solaredge/
+"""
+
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.light import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_API_KEY, CONF_MONITORED_VARIABLES, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['solaredge==0.0.2']
+
+DOMAIN = "solaredge"
+
+# Config for solaredge monitoring api requests.
+CONF_SITE_ID = "site_id"
+
+UPDATE_DELAY = timedelta(minutes=10)
+SCAN_INTERVAL = timedelta(minutes=10)
+
+# Supported sensor types:
+# Key: ['json_key', 'name', unit, icon]
+SENSOR_TYPES = {
+    'life_time_data': ['lifeTimeData', "Lifetime energy", 'Wh',
+                       'mdi:solar-power'],
+    'last_year_data': ['lastYearData', "Energy this year", 'Wh',
+                       'mdi:solar-power'],
+    'last_month_data': ['lastMonthData', "Energy this month", 'Wh',
+                        'mdi:solar-power'],
+    'last_day_data': ['lastDayData', "Energy today", 'Wh',
+                      'mdi:solar-power'],
+    'current_power': ['currentPower', "Current Power", 'W',
+                      'mdi:solar-power']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SITE_ID): cv.string,
+    vol.Optional(CONF_NAME, default='SolarEdge'): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=['current_power']):
+    vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create the SolarEdge Monitoring API sensor."""
+    import solaredge
+    from requests.exceptions import HTTPError, ConnectTimeout
+
+    api_key = config[CONF_API_KEY]
+    site_id = config[CONF_SITE_ID]
+    platform_name = config[CONF_NAME]
+
+    _LOGGER.debug("Setting up SolarEdge Monitoring API")
+
+    # Create new SolarEdge object to retrieve data
+    api = solaredge.Solaredge(api_key)
+
+    # Check if api can be reached and site is active
+    try:
+        response = api.get_details(site_id)
+
+        if response['details']['status'].lower() != 'active':
+            _LOGGER.debug("SolarEdge site is not active")
+            return False
```

Comment:
Nothing is checking this return value. Just `return`.
```suggestion
            return
```